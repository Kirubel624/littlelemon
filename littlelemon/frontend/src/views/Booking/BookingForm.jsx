import React, { useEffect, useMemo, useState } from 'react';
import {LeftCircleFilled} from '@ant-design/icons'
import { message } from 'antd';

const BookingForm = ({ handleReservation, availableTimes, updateAvailableTimes, date, setDate }) => {
  const [time, setTime] = useState("17:00");
  const [numberOfGuests, setNumberOfGuests] = useState(1);
  const [occasion, setOccasion] = useState("Anniversary");
  const [bookingInfo, setBookingInfo] = useState({
    userID:"6cftgs6uegtfsty",
    reservationDate: date,
    reservationTime: "17:00",
    numberOfGuests: "1",
    occassion: "Anniversary",
  });

 
useEffect(()=>{
setTime(availableTimes[0])
console.log(availableTimes.length)
},[availableTimes])
  // console.log("Booking information", bookingInfo);
  const memoizedBookingInfo = useMemo(() => ({
    userID: "awhjfhjawfawdawd",
    reservationDate: date,
    reservationTime: time,
    numberOfGuests: numberOfGuests,
    occassion: occasion
  }), [date, time, numberOfGuests, occasion]);
  
  useEffect(() => {
    const today = new Date().toISOString().split('T')[0];
    if (date < today) {
      setDate(today);
    }
    setBookingInfo(memoizedBookingInfo);
    console.log(memoizedBookingInfo);
  }, [date, setDate, memoizedBookingInfo]);

  return (
    <div className='flex flex-col mt-24'>
    <span className='flex justify-center items-center   -red-900'>
  <LeftCircleFilled className='block md:hidden absolute left-10 text-xl text-[#495E57]'/>
   <h1 className='font-karla text-center font-extrabold text-xl'>RESERVE A TABLE</h1></span>
      <form className='flex flex-col justify-center items-center'
       onSubmit={(e)=>handleReservation(e,bookingInfo)}>
        <label htmlFor="res-date">Choose date</label>
        <input
          type="date"
          id="res-date"
          name="res-date"
          value={date}
          onChange={(e) => {
            setDate(e.target.value);
            updateAvailableTimes(e.target.value);
          }}
          min={new Date().toISOString().split('T')[0]}
          aria-label="Choose date for reservation"
        />
        <label htmlFor="res-time">Choose time</label>
       {availableTimes?.length>0? <select 
          name="res-time"
       
       id="res-time" value={time} onChange={(e) => { setTime(e.target.value) }} aria-label="Choose time for reservation">
          {availableTimes?.map((time) => (
            <option key={time} value={time}>
              {time}
            </option>
          ))}
        </select>:<p>No more booking times available on this date. Pick another date</p>}
        <label htmlFor="guests">Number of guests</label>
        <input
          type="number"
          placeholder="1"
          min="1"
          max="10"
          id="guests"
          value={numberOfGuests}
          onChange={(e) => { setNumberOfGuests(e.target.value) }}
          aria-label="Enter the number of guests"
        />
        <label htmlFor="occasion">Occasion</label>
        <select id="occasion" value={occasion} onChange={(e) => setOccasion(e.target.value)} aria-label="Choose occasion for reservation">
          <option value="Birthday">Birthday</option>
          <option value="Anniversary">Anniversary</option>
          <option value="Engagement">Engagement</option>
          <option value="Graduation">Graduation</option>
        </select>
        <button type="submit" aria-label="Submit reservation form">
        Make Your reservation
        </button>
      </form>
    </div>
  );
};

export default BookingForm;
