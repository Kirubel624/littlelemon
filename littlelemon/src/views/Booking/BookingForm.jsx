import React, { useEffect, useState } from 'react';

const BookingForm = ({ availableTimes, updateAvailableTimes }) => {
  const [date, setDate] = useState("00/00/0000");
  const [time, setTime] = useState("00:00");
  const [numberOfGuests, setNumberOfGuests] = useState(1);
  const [occasion, setOccasion] = useState("Anniversary");
  const [bookingInfo, setBookingInfo] = useState({
    date: "00/00/0000",
    time: "00:00",
    numberOfGuests: "1",
    occasion: "",
  });

  const handleReservation = (e) => {
    e.preventDefault();
    console.log(date, time, numberOfGuests, occasion);
    setBookingInfo((prevBooking) => ({ ...prevBooking, date: date, time: time, numberOfGuests: numberOfGuests, occasion: occasion }));
    // setDate("");
    // setTime("");
    // setNumberOfGuests("");
    // setOccasion("");
  };

  console.log("Booking information", bookingInfo);

  useEffect(() => {
    const today = new Date().toISOString().split('T')[0];
    if (date < today) {
      setDate(today);
    }
  }, [date, setDate]);

  return (
    <>
      <form className='pt-24 flex flex-col' onSubmit={handleReservation}>
        <label htmlFor="res-date">Choose date</label>
        <input
          type="date"
          id="res-date"
          value={date}
          onChange={(e) => {
            setDate(e.target.value);
            updateAvailableTimes(e.target.value);
          }}
          min={new Date().toISOString().split('T')[0]}
          aria-label="Choose date for reservation"
        />
        <label htmlFor="res-time">Choose time</label>
        <select id="res-time" value={time} onChange={(e) => { setTime(e.target.value) }} aria-label="Choose time for reservation">
          {availableTimes.map((time) => (
            <option key={time} value={time}>
              {time}
            </option>
          ))}
        </select>
        <label htmlFor="guests">Number of guests</label>
        <input
          type="number"
          placeholder="1"
          min="1"
          max="10"
          id="guests"
          value={numberOfGuests}
          onChange={(e) => { setNumberOfGuests(e.target.value) }}
          aria-label="Enter the number of guests"
        />
        <label htmlFor="occasion">Occasion</label>
        <select id="occasion" value={occasion} onChange={(e) => setOccasion(e.target.value)} aria-label="Choose occasion for reservation">
          <option value="Birthday">Birthday</option>
          <option value="Anniversary">Anniversary</option>
          <option value="Engagement">Engagement</option>
          <option value="Graduation">Graduation</option>
        </select>
        <input type="submit" value="Make Your reservation" aria-label="Submit reservation form" />
      </form>
    </>
  );
};

export default BookingForm;
