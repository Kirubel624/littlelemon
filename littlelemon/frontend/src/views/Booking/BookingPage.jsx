import React, { useEffect, useReducer, useState } from 'react';
import BookingForm from './BookingForm';
import {message} from 'antd'
import { useNavigate } from 'react-router-dom';
export const updateTimesReducer = (state, action) => {
  switch (action?.type) {
    case "UPDATE_TIMES":
      return action?.payload?.updateTimes;
    default:
      return state;
  }
};
export const initializeTimes = (times) => {
  return times;
};

const BookingPage = () => {
  const navigate=useNavigate()
  const dateNow=new Date().toISOString().split('T')[0]
  const [times, setTimes] = useState([]);

  const [date, setDate] = useState(dateNow);
  const [availableTimes, dispatch] = useReducer(updateTimesReducer,[],initializeTimes);
  const handleReservation = (e,bookingInfo) => {
    e.preventDefault();
    console.log(bookingInfo,"inside handleReservation booking info")
  
    fetch('http://localhost:8000/api/v1/reserve',{
      method: 'POST',
      headers:{
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(bookingInfo)
    }).then((res)=>{
      console.log(res,"res in handle reservation")
      if(res.status===409&&availableTimes.length!==0){
        message.info('Time has been booked for this date!')

      }else if(res.status===201){
        // message.success("Reservation successful!")
         navigate(`/bookingConfirmation?date=${bookingInfo.reservationDate}&occassion=${bookingInfo.occassion}&time=${bookingInfo.reservationTime}`)

        
      }else if(availableTimes.length===0){
        message.info('All times for this date have been booked!. Try another date')

      }
      res.json()
       updateAvailableTimes(date)
    })
    // setDate("");
    // setTime("");
    // setNumberOfGuests("");
    // setOccasion("");

  };


  const fetchData = async (date) => {
    try {
      const response = await fetch(`http://localhost:8000/api/v1/reserve/getAvailableDates?date=${date}`);
      const data = await response.json();
      console.log(data.data.availableTimes);
        setTimes(data.data.availableTimes);
      
    } catch (err) {
      console.log(err);
    }
  };

  const updateAvailableTimes = (date) => {
    setDate(date);
    fetchData(date);
  };

  useEffect(() => {
    fetchData(date);
  }, [date]);

  useEffect(() => {
    dispatch({ type: "UPDATE_TIMES", payload: { updateTimes: times } });
initializeTimes(times)
  }, [times]);
console.log(initializeTimes(times),"function initialize times")
  return (
    <>
      <BookingForm handleReservation={handleReservation} availableTimes={availableTimes} updateAvailableTimes={updateAvailableTimes} date={date} setDate={setDate} />
    </>
  );
};

export default BookingPage;
