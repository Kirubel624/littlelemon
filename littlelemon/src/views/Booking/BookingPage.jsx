import React, { useEffect, useReducer, useState } from 'react';
import BookingForm from './BookingForm';
import {Tabs, message} from 'antd'
import { useNavigate } from 'react-router-dom';
import Bookings from './Bookings';
export const updateTimesReducer = (state, action) => {
  switch (action?.type) {
    case "UPDATE_TIMES":
      return action?.payload?.updateTimes;
    default:
      return state;
  }
};
export const initializeTimes = (times) => {
  return times;
};

const BookingPage = () => {
  const navigate=useNavigate()
  const dateNow=new Date().toISOString().split('T')[0]
  const [times, setTimes] = useState([]);
  const [activeKey, setActiveKey] = useState("reserve");

  const [date, setDate] = useState(dateNow);
  const [availableTimes, dispatch] = useReducer(updateTimesReducer,[],initializeTimes);
  const[timeData, setTimeData] = useState()
  const handleReservation = (e,bookingInfo) => {
    e.preventDefault();
    console.log(bookingInfo,"inside handleReservation booking info")
  
    fetch('http://localhost:8000/api/v1/reserve',{
      method: 'POST',
      headers:{
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(bookingInfo)
    }).then((res)=>{
      console.log(res,"res in handle reservation")
      if(res.status===409&&availableTimes.length!==0){
        message.info('Time has been booked for this date!')

      }else if(res.status===201){
        // message.success("Reservation successful!")
         navigate(`/bookingConfirmation?date=${bookingInfo.reservationDate}&occassion=${bookingInfo.occassion}&time=${bookingInfo.reservationTime}`)

        
      }else if(availableTimes.length===0){
        message.info('All times for this date have been booked!. Try another date')

      }
      res.json()
       updateAvailableTimes(date)
    })
    // setDate("");
    // setTime("");
    // setNumberOfGuests("");
    // setOccasion("");

  };


  const fetchData = async (date) => {
    try {
      const response = await fetch(`http://localhost:8000/api/v1/reserve/getAvailableDates?date=${date}`);
      const data = await response.json();
      console.log(data.data.availableTimes);
        setTimes(data.data.availableTimes);
          setTimeData(data.data);
        console.log(data.data,"data data data&&&&&&&&&&&&&&");
    
    } catch (err) {
      console.log(err);
    }
  };

  const updateAvailableTimes = (date) => {
    setDate(date);
    fetchData(date);
  };
console.log(timeData,"&&&&&&&&&&&&tiem data")
  useEffect(() => {
    fetchData(date);
  }, [date]);

  useEffect(() => {
    dispatch({ type: "UPDATE_TIMES", payload: { updateTimes: times } });
initializeTimes(times)
setTimeData(timeData)
  }, [times]);
console.log(initializeTimes(times),"function initialize times")
  const handleTabChange = (key) => {
    setActiveKey(key);
  };
 const booking = [
   {
     key: "reservations",
     label: "Reservations",

     children: (
       <Bookings
         date={date}
         setDate={setDate}
         availableTimesT={availableTimes}
         updateAvailableTimes={updateAvailableTimes}
       />
     ),
   },
   {
     key: "reserve",
     label: "Reserve",
     children: (
       <BookingForm
         handleReservation={handleReservation}
         availableTimesT={availableTimes}
         timeData={timeData}
         updateAvailableTimes={updateAvailableTimes}
         date={date}
         setDate={setDate}
       />
     ),
   },
 ];
return (
  <>
    <Tabs
      className="w-full mt-24 pl-10"
      items={booking}
      activeKey={activeKey}
      onChange={handleTabChange}></Tabs>
  </>
);
};

export default BookingPage;
