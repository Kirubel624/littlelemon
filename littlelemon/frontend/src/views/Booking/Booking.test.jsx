import {fireEvent, getByLabelText, getByText, render, screen} from '@testing-library/react'

import BookingForm from './BookingForm'
import { updateTimesReducer, initializeTimes } from './BookingPage';  // Import the function from your BookingPage
import HomePage from '../Home/index'
import { describe, expect, it, vi, afterEach, beforeEach } from 'vitest'
import TestRenderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import * as utils from './BookingPage'

describe("Booking Form",()=>{
    test("Static text test",()=>{
    // const initializeTimes = jest.fn()
    render(<BookingForm availableTimes={["17:00","18:00","19:00","20:00","21:00"]}/>)

    const textElement=screen.getByText(/Make Your reservation/i)
console.log(textElement)
    expect(textElement).toBeInTheDocument()
    // expect(initializeTimes).toEqual(["17:00","18:00","19:00","20:00","21:00"])
}
)

it("Will create a snapshot", ()=>{
    const homePage=TestRenderer.create(<MemoryRouter><HomePage/></MemoryRouter>).toJSON()
    expect(homePage).toMatchSnapshot()
})
it("Returns available times",()=>{
  const times=["17:00","18:00","19:00","20:00","21:00"]
  const result = utils.initializeTimes(times)
  const expectedTimes=["17:00","18:00","19:00","20:00","21:00"]
  expect(result).toEqual(expectedTimes)
})
it('Returns the same value that is provided in the state', () => {
    const initialState = ['17:00', '18:00', '19:00', '20:00', '21:00'];
    const action = { type: 'UPDATE_TIMES', payload: {  updateTimes: ['17:00', '18:00', '19:00', '20:00', '21:00'] } };

    // Dispatch the action and get the updated state
    const updatedState = updateTimesReducer(initialState, action);

    // Assert that the updated state is the same as the initial state
    expect(updatedState).toStrictEqual(initialState);
  });
it("User can make reservations",()=>{
    const handleReservation=jest.fn()
    const { getByLabelText, getByText } = render(<BookingForm availableTimes={["17:00","18:00","19:00","20:00","21:00"]} handleReservation={handleReservation}/>);
  
    const dateInput = getByLabelText('Choose date');
    const timeSelect = getByLabelText('Choose time');
    const submitButton = getByText('Make Your reservation');
    fireEvent.change(dateInput, { target: { value: '2023-12-08' } });
    fireEvent.change(timeSelect, { target: { value: '17:00' } });
    fireEvent.click(submitButton);
    expect(handleReservation).toHaveBeenCalled()
})
})