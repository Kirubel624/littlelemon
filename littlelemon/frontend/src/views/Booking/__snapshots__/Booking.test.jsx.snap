// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Booking Form > Will create a snapshot 1`] = `
<main
  className="pt-20"
>
  <section
    className="bg-[#495E57] w-full flex justify-between -red-900 items-center px-[20vw] py-10"
  >
    <div
      className="lg:w-[25vw] w-full  -green-900 flex flex-col justify-center item"
    >
      <h1
        className=" font-markazi text-[3.25rem] leading-[0.8] text-[#F4CE14]"
      >
        Little Lemon
      </h1>
      <p
        className="text-white font-markazi text-3xl pb-4"
      >
        Chicago
      </p>
      <p
        className="text-white font-karla pb-8"
      >
        We are a family-owned Mediterranean restaurant, focused on traditional recipes with a modern twist
      </p>
      <a
        aria-label="Reserve a table"
        className="bg-[#F4CE14] text-black px-4 py-2 rounded text-center"
        href="/booking"
        onClick={[Function]}
      >
        Reserve a table
      </a>
    </div>
    <div
      className="static  -red-900 flex justify-center items-"
    >
      <img
        alt="Little Lemon Restaurant"
        aria-hidden="true"
        className="lg:block hidden w-[300px] rounded-xl object-cover h-[335px] absolute right-[20vw] top-[8rem]"
        src="https://res.cloudinary.com/dvqawl4nw/image/upload/v1701453328/pbygrd9a1r6puekc3x2y.jpg"
      />
    </div>
  </section>
  <section
    className="flex flex-col justify-center items- py-24 px-[20vw] borer borer-green-900"
  >
    <div
      className="flex md:flex-row flex-col justify-between md:items-center mb-4 borer borer-green-900"
    >
      <h1
        className=" font-markazi text-5xl md:pb-0 pb-4"
      >
        This week's specials!
      </h1>
      <button
        className="bg-[#F4CE14] text-black px-4 py-2 rounded"
      >
        Order online
      </button>
    </div>
    <div
      className="flex flex-wrap justify-between items-start borer borer-green-900"
    >
      <div>
        <article
          aria-label="Card for Greek Salad"
          className="flex flex-col justify-start item-start mb-8 w-full lg:w-[230px] boder boder-red-900
     bg-gray-200 rounded-[16px] font-karla hover:cursor-pointer"
        >
          <div
            className="borer rounded-t-xl borer-red-800 overflow-clip"
          >
            <img
              alt="Greek Salad"
              className="object-cover duration-300 delay=150 transition-all ease-in hover:scale-110 w-full lg:w-[250px] h-[224px] "
              src="https://res.cloudinary.com/dvqawl4nw/image/upload/v1701456411/egxxnqvfc50gfksic7dk.png"
            />
          </div>
          <span
            className="flex items-center justify-between"
          >
             
            <p
              aria-label="Greek Salad name"
              className="p-4"
            >
              Greek Salad
            </p>
            <p
              className="p-4 text-[#EE9972]"
            >
              $12.99
            </p>
          </span>
          <p
            aria-label="Greek Salad description"
            className="p-4"
          >
            A refreshing and healthy salad inspired by Mediterranean cuisine. 
      This Greek Salad features a mix of crisp vegetables like tomatoes, cucumbers, 
      and bell peppers, topped with feta cheese ...
          </p>
          <button
            className="bg-none"
          >
            <p
              aria-label="Order a delivery for Greek Salad"
              className="flex p-4"
            >
              Order a delivery
              <img
                className="ml-6"
                src="/src/assets/Delivery.svg"
              />
            </p>
          </button>
        </article>
      </div>
      <div>
        <article
          aria-label="Card for Bruschetta"
          className="flex flex-col justify-start item-start mb-8 w-full lg:w-[230px] boder boder-red-900
     bg-gray-200 rounded-[16px] font-karla hover:cursor-pointer"
        >
          <div
            className="borer rounded-t-xl borer-red-800 overflow-clip"
          >
            <img
              alt="Bruschetta"
              className="object-cover duration-300 delay=150 transition-all ease-in hover:scale-110 w-full lg:w-[250px] h-[224px] "
              src="https://res.cloudinary.com/dvqawl4nw/image/upload/v1701454943/jbcggkchmsfgnomtobkp.jpg"
            />
          </div>
          <span
            className="flex items-center justify-between"
          >
             
            <p
              aria-label="Bruschetta name"
              className="p-4"
            >
              Bruschetta
            </p>
            <p
              className="p-4 text-[#EE9972]"
            >
              $5.99
            </p>
          </span>
          <p
            aria-label="Bruschetta description"
            className="p-4"
          >
            A classic Italian appetizer, 
      Bruschetta consists of grilled bread rubbed with garlic and topped with diced tomatoes,
       fresh basil, and drizzled with balsamic glaze. It offers a delightful...
          </p>
          <button
            className="bg-none"
          >
            <p
              aria-label="Order a delivery for Bruschetta"
              className="flex p-4"
            >
              Order a delivery
              <img
                className="ml-6"
                src="/src/assets/Delivery.svg"
              />
            </p>
          </button>
        </article>
      </div>
      <div>
        <article
          aria-label="Card for Lemon Dessert"
          className="flex flex-col justify-start item-start mb-8 w-full lg:w-[230px] boder boder-red-900
     bg-gray-200 rounded-[16px] font-karla hover:cursor-pointer"
        >
          <div
            className="borer rounded-t-xl borer-red-800 overflow-clip"
          >
            <img
              alt="Lemon Dessert"
              className="object-cover duration-300 delay=150 transition-all ease-in hover:scale-110 w-full lg:w-[250px] h-[224px] "
              src="https://res.cloudinary.com/dvqawl4nw/image/upload/v1701454934/p3gvo4pew7extg8johdn.jpg"
            />
          </div>
          <span
            className="flex items-center justify-between"
          >
             
            <p
              aria-label="Lemon Dessert name"
              className="p-4"
            >
              Lemon Dessert
            </p>
            <p
              className="p-4 text-[#EE9972]"
            >
              $5.00
            </p>
          </span>
          <p
            aria-label="Lemon Dessert description"
            className="p-4"
          >
            A zesty and indulgent lemon dessert that satisfies the sweet tooth.
       This dessert may include lemon bars, lemon tarts, or lemon-flavored cakes. 
       The bright and citrusy flavor of lemon add...
          </p>
          <button
            className="bg-none"
          >
            <p
              aria-label="Order a delivery for Lemon Dessert"
              className="flex p-4"
            >
              Order a delivery
              <img
                className="ml-6"
                src="/src/assets/Delivery.svg"
              />
            </p>
          </button>
        </article>
      </div>
    </div>
  </section>
  <section
    className="bg-[#495E57] py-24 px-[20vw]"
  >
    <h1
      aria-label="Testimonials"
      className="font-markazi text-5xl text-center text-white"
    >
      Testimonials
    </h1>
    <div
      className="grid md:grid-cols-2 grid-cols-1 xl:grid-cols-4 gap-4 justify-around items-center pt-12"
    >
      <article
        aria-label="Testimonial from Amy"
        className="flex flex-col bg-white mb-4 rounded-[16px] font-karla"
      >
        <p
          aria-label="Rating: 5 stars"
          className="pt-8 pl-8 pb-4"
        >
          Rating: 
          5
        </p>
        <p
          className="flex items-center pl-8"
        >
          <img
            alt="Profile image of Amy"
            className="w-14 rounded-full h-14 object-cover mr-4"
            src="https://res.cloudinary.com/dvqawl4nw/image/upload/v1700576200/lupkrg0d6s9zksiqg9fg.avif"
          />
          Amy
        </p>
        <p
          aria-label="Testimonial: Great food and hospitality"
          className="pt-4 pb-8 px-8"
        >
          Great food and hospitality
        </p>
      </article>
      <article
        aria-label="Testimonial from Maria"
        className="flex flex-col bg-white mb-4 rounded-[16px] font-karla"
      >
        <p
          aria-label="Rating: 4 stars"
          className="pt-8 pl-8 pb-4"
        >
          Rating: 
          4
        </p>
        <p
          className="flex items-center pl-8"
        >
          <img
            alt="Profile image of Maria"
            className="w-14 rounded-full h-14 object-cover mr-4"
            src="https://res.cloudinary.com/dvqawl4nw/image/upload/v1700576198/z8kjqwsl82lthhhdljc2.avif"
          />
          Maria
        </p>
        <p
          aria-label="Testimonial: Great food and hospitality"
          className="pt-4 pb-8 px-8"
        >
          Great food and hospitality
        </p>
      </article>
      <article
        aria-label="Testimonial from Silvia"
        className="flex flex-col bg-white mb-4 rounded-[16px] font-karla"
      >
        <p
          aria-label="Rating: 4.3 stars"
          className="pt-8 pl-8 pb-4"
        >
          Rating: 
          4.3
        </p>
        <p
          className="flex items-center pl-8"
        >
          <img
            alt="Profile image of Silvia"
            className="w-14 rounded-full h-14 object-cover mr-4"
            src="https://res.cloudinary.com/dvqawl4nw/image/upload/v1700576196/f2uibet1zsaah1eezqit.avif"
          />
          Silvia
        </p>
        <p
          aria-label="Testimonial: Great food and hospitality"
          className="pt-4 pb-8 px-8"
        >
          Great food and hospitality
        </p>
      </article>
      <article
        aria-label="Testimonial from John"
        className="flex flex-col bg-white mb-4 rounded-[16px] font-karla"
      >
        <p
          aria-label="Rating: 4.7 stars"
          className="pt-8 pl-8 pb-4"
        >
          Rating: 
          4.7
        </p>
        <p
          className="flex items-center pl-8"
        >
          <img
            alt="Profile image of John"
            className="w-14 rounded-full h-14 object-cover mr-4"
            src="https://res.cloudinary.com/dvqawl4nw/image/upload/v1701541341/vlad1gswy3vlxentw9al.jpg"
          />
          John
        </p>
        <p
          aria-label="Testimonial: Great food and hospitality"
          className="pt-4 pb-8 px-8"
        >
          Great food and hospitality
        </p>
      </article>
    </div>
  </section>
  <section
    aria-label="About Little Lemon in Chicago"
    className="py-24 px-[20vw] flex"
  >
    <div
      className="pr-6"
    >
      <h1
        aria-label="Little Lemon"
        className="font-markazi text-[3.25rem] leading-[0.8]"
      >
        Little Lemon
      </h1>
      <p
        className="font-markazi text-3xl pb-4"
      >
        Chicago
      </p>
      <p
        aria-label="Description about Little Lemon in Chicago"
        className="font-karla"
      >
        Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit. Exercitation veniam consequat sunt nostrud amet. Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit.
      </p>
    </div>
    <div
      aria-hidden="true"
      className="w-"
    >
      <img
        alt="Little Lemon Chicago"
        className="hidden lg:block object-cover h-full"
        src="https://res.cloudinary.com/dvqawl4nw/image/upload/v1701684970/jnatcrxa3sxjppcjpeeh.png"
      />
    </div>
  </section>
</main>
`;
